contexts/LedgerTransportContext.tsx

```diff
@@ -2,47 +2,70 @@ import { createContext, ReactNode, useCallback, useState } from "react";
 import Transport from "@ledgerhq/hw-transport";
 import TransportWebHid from "@ledgerhq/hw-transport-webhid";
 import { captureException } from "@sentry/react";
+import KaspaApp from "hw-app-kaspa";
 
 interface LedgerTransportContextType {
   transport: Transport | null;
   connect: () => Promise<void>;
   disconnect: () => Promise<void>;
   isConnecting: boolean;
+  isAppOpen: boolean;
 }
 
 export const LedgerTransportContext = createContext<
   LedgerTransportContextType | undefined
 >(undefined);
 
+async function checkKaspaAppOpen(transport: Transport) {
+  const kaspaApp = new KaspaApp(transport);
+  try {
+    await kaspaApp.getVersion();
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
 export function LedgerTransportProvider({ children }: { children: ReactNode }) {
   const [transport, setTransport] = useState<Transport | null>(null);
   const [isConnecting, setIsConnecting] = useState(false);
+  const [isAppOpen, setIsAppOpen] = useState(false);
+  const [isAppOpenCheckInterval, setIsAppOpenCheckInterval] =
```

Comment:
Interesting `setIsAppOpenCheckInterval` function is never called?

Reply:
My bad, I forgot to remove it.

Reply:
https://github.com/forbole/kastle/pull/41/commits/20a76c49d0ccbe2481443a4df38aa559bb31f3f3